services/database/models/HackModel.js

```diff
@@ -0,0 +1,15 @@
+import mongoose from "mongoose";
+
+const Schema = mongoose.Schema;
+const hackSchema = new Schema({
+  title: { type: String, default: "" },
+  description: { type: String, default: "" },
+  goal: { type: String, default: "" },
+  team: { type: [Schema.Types.ObjectId], default: [] },
```

Comment:
should this be defined as

```
team: { type: [Schema.Types.ObjectId], ref: 'User', default: [] },
```

(and the other two)

Reply:
That's what I originally thought as well, appears the team have changed their approach, that's why I used populate below to refer to the model:

```javascript
  let result = await Hack.findById(id)
      .select("-likes")
      .populate("team", "-email", User);
```

https://stackoverflow.com/questions/14627948/mongoose-populate-field-without-ref-option